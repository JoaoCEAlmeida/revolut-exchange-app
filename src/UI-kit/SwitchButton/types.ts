export interface StyledButtonProps {
  transactionType: string
}