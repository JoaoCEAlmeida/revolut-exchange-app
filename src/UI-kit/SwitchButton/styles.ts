import styled from 'styled-components';
import { StyledButtonProps } from './types';

export const StyledSwitchButton = styled.div<StyledButtonProps>`
  position: absolute;
  align-items: center;
  display: flex;
  justify-content: center;
  top: 50%;
  left: 50%;
  border-radius: 50%;
  border: none;
  background-color: ${(props) => props.theme.colors.background.main};
  width: 40px;
  height: 40px;
  cursor: pointer;
  transform: translate(-50%, -50%);

  svg {
    transition: all 0.2s;
    transform: ${(props) => (props.transactionType === 'Buy' ? 'rotateX(180deg)' : '')};
    color: ${(props) => props.theme.colors.primary.main};
  }
`;
