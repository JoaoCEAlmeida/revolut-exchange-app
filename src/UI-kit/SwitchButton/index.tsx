import React from 'react';
import { ArrowDownIcon } from '@heroicons/react/outline';

import { StyledSwitchButton } from './styles';
import { StyledButtonProps } from './types';

const SwitchButton: React.FC<StyledButtonProps> = ({ transactionType, onClick, ...rest }) => {
  return (
    <StyledSwitchButton
      data-testid="test-switchButton"
      transactionType={transactionType}
      onClick={onClick}
      {...rest}>
      <ArrowDownIcon height="20px" />
    </StyledSwitchButton>
  );
};

export default SwitchButton;
