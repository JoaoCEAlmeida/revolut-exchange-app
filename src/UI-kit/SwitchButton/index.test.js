import { screen, fireEvent } from '@testing-library/react';
import SwitchButton from './index';
import renderWithProviders from '../../test/providers';

const defaultProps = {
  transactionType: 'Sell',
  onClick: jest.fn()
};

describe('SwitchButton Tests', () => {
  test('should render correctly', () => {
    const { SwitchButtonTest } = renderWithProviders(() => <SwitchButton {...defaultProps} />);

    expect(SwitchButtonTest).toMatchSnapshot();
  });

  test('should fire the onClick event', () => {
    const clickHandler = jest.fn();

    renderWithProviders(() => <SwitchButton {...defaultProps} onClick={clickHandler} />);

    fireEvent.click(screen.getByTestId('test-switchButton'));

    expect(clickHandler).toHaveBeenCalled();
  });
});
