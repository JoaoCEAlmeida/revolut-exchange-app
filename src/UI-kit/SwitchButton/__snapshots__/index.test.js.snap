// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Tests should render correctly 1`] = `undefined`;

exports[`SwitchButton Tests should render correctly 1`] = `undefined`;
